fixtures/attribute-behavior/AttributeTableSnapshot.md

```diff
@@ -11980,21 +11980,21 @@
 | `value=(empty string)`| (initial)| `<empty string>` |
 | `value=(array with string)`| (changed)| `"string"` |
 | `value=(empty array)`| (initial)| `<empty string>` |
-| `value=(object)`| (changed, ssr error, ssr mismatch)| `"result of toString()"` |
+| `value=(object)`| (changed)| `"result of toString()"` |
 | `value=(numeric string)`| (changed)| `"42"` |
 | `value=(-1)`| (changed)| `"-1"` |
 | `value=(0)`| (changed)| `"0"` |
 | `value=(integer)`| (changed)| `"1"` |
 | `value=(NaN)`| (changed, warning)| `"NaN"` |
 | `value=(float)`| (changed)| `"99.99"` |
-| `value=(true)`| (changed, ssr mismatch)| `"true"` |
-| `value=(false)`| (changed, ssr mismatch)| `"false"` |
+| `value=(true)`| (changed)| `"true"` |
+| `value=(false)`| (changed)| `"false"` |
 | `value=(string 'true')`| (changed)| `"true"` |
 | `value=(string 'false')`| (changed)| `"false"` |
 | `value=(string 'on')`| (changed)| `"on"` |
 | `value=(string 'off')`| (changed)| `"off"` |
-| `value=(symbol)`| (initial, warning)| `<empty string>` |
-| `value=(function)`| (initial, warning)| `<empty string>` |
+| `value=(symbol)`| (initial, warning, ssr error, ssr mismatch)| `<empty string>` |
+| `value=(function)`| (initial, warning, ssr mismatch)| `<empty string>` |
```

Comment:
Should this be fixed? https://github.com/facebook/react/pull/24077 fixes these ssr mismatches for `<select value />`. Can we leverage https://github.com/facebook/react/pull/24077 to fix this new ssr mismatch as well as the (existing) ssr mismatch for `<option value={symbol} />`?

Reply:
What tag is this on? The table doesn't say.

Reply:
One thing that's tricky is where we do these conversions. Should they be done by the `select` tag so that if you provide no option, it should happen anyway. Or should they be done by the `option` tag when the comparison actually happens.

It probably makes more sense to do them on the `select` and eagerly convert them to a cloned array of strings.

Reply:
> What tag is this on? The table doesn't say.

The expand feature on github.com is broken here. This line is under the "`value` (on `<textarea>` inside `<div>`)" section: https://github.dev/facebook/react/pull/24083#pullrequestreview-909231364

Reply:
I see so the change happened in https://github.com/facebook/react/pull/23359/commits/e915eee9fcc56cfbfe42f7e4b50d6cf3b9db9c6b

The thing is that the fix in #24077 kind of relies on us only toStringing if one if the sides is not defined. I think that the proper fix is to always toString if a value is provided (and always doing so inside the component it is provided so in the `select` instead of the option for selected value). In other words, if we fix that then maybe we'd actually revert to erroring again.

So I think that it's correct that this errors.

The bug is that the textarea on the client doesn't error by going through getToStringValue. In other words, I think we should revert https://github.com/facebook/react/pull/13389

Reply:
I guess we pretty pervasively check for symbols and silently ignore them for custom props too, which is not great. We should at least add warnings for all those cases and then change it along with other DOM breaking changes we have planned.

I'm not sure if it's worth fixing on the server case. Might as well sneak in a small breakage that we're planning on doing anyway.